### PR TITLE
Add support of ReduceSum int64

### PR DESCRIPTION
**Pull request by yufenglee**
**Description**: Describe your changes.

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

**Comment by hariharans29** [comment]
I think it is better to reference the issue that this is trying to resolve so that the poster gets some sort of notification :). In this case, resolving #1657

File: onnxruntime/core/providers/cpu/reduction/reduction_ops.cc [hunk 1]
```diff
@@ -48,6 +48,7 @@ REGISTER_UNARY_ELEMENTWISE_KERNEL(ReduceMean, 1);
 REGISTER_UNARY_ELEMENTWISE_KERNEL(ReduceMin, 1);
 REGISTER_UNARY_ELEMENTWISE_KERNEL(ReduceProd, 1);
 REGISTER_UNARY_ELEMENTWISE_KERNEL(ReduceSum, 1);
+REGISTER_UNARY_ELEMENTWISE_KERNEL_INT64_ONLY(ReduceSum, 1);
```
Comment (hariharans29) [inline onnxruntime/core/providers/cpu/reduction/reduction_ops.cc]: Safe to have a  corresponding test as well.

**Comment by yufenglee** [comment]
> I think it is better to reference the issue that this is trying to resolve so that the poster gets some sort of notification :). In this case, resolving #1657

Actually didn't notice this issue. This is a request from 1st party model.

**Comment by hariharans29** [comment]
Ah sorry about that - I thought of resolving that issue and I saw your PR, so automatically assumed you were resolving that issue- turns out that the issue poster closed it...

**Pull request merged**